components/auth/AuthForm.tsx

```diff
@@ -0,0 +1,229 @@
+          });
+        }
+      }
+    },
+  };
+  const IdRegister = auth === "signin" ? IdRegforSignin : IdRegforSignup;
+  const passwordRegForSignin = {
+    required: "비밀번호를 입력해주세요",
+  };
+  const passwordRegForSignup = {
+    required: "비밀번호를 입력해주세요",
+    pattern: {
+      value: /^(?=.*[a-zA-Z])(?=.*[0-9]).{8,}$/,
+      message: "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요",
+    },
+  };
+  const passwordRegister =
+    auth === "signin" ? passwordRegForSignin : passwordRegForSignup;
+  const passCheckRegister = {
+    onBlur: (e: FocusEvent<HTMLInputElement>) => {
+      if (watch("password") !== e.target.value) {
+        setError("passwordCheck", {
+          type: "NotSame",
+          message: "비밀번호가 다릅니다",
+        });
+      } else {
+        clearErrors("passwordCheck");
+      }
+    },
+  };
+  const passPlaceholder =
+    auth === "signin"
+      ? "비밀번호를 입력해주세요"
+      : "영문, 숫자를 조합해 8자 이상 입력해 주세요";
+  const authType = auth === "signin" ? "로그인" : "회원가입";
+
+  const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    if (auth === "signin") {
+      try {
+        await tryUserCheck(data);
+      } catch (err) {
+        setError("password", {
+          type: "checkPassword",
+          message: "비밀번호를 확인해주세요",
+        });
+        setError("Id", {
+          type: "CheckId",
+          message: "아이디를 확인해주세요",
+        });
+      }
+    } else {
+      try {
+        await signUpApi(data);
+      } catch (err) {
+        console.log(err);
+      }
+    }
+  };
+
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      <Label htmlFor="Id">아이디</Label>
+      <Div>
+        <Input
+          id="Id"
+          placeholder="이메일을 입력해주세요"
+          type="text"
+          className={errors.Id ? "err" : ""}
+          aria-invalid={errors.Id ? "true" : "false"}
+          {...register("Id", IdRegister)}
+        />
+        {errors.Id && <ErrorMessage>{errors?.Id?.message}</ErrorMessage>}
+      </Div>
+      <Label htmlFor="password">비밀번호</Label>
+      <Div>
+        <Input
+          id="password"
+          placeholder={passPlaceholder}
+          type={passShow ? "text" : "password"}
+          className={errors.password ? "err" : ""}
+          aria-invalid={errors.password ? "true" : "false"}
+          {...register("password", passwordRegister)}
+        />
+        <AuthButton target={setPassShow} />
+        {errors.password && (
+          <ErrorMessage>{errors?.password?.message}</ErrorMessage>
+        )}
+      </Div>
+      {auth === "signup" && (
+        <>
+          <Label htmlFor="passwordCheck">비밀번호 확인</Label>
+          <Div>
+            <Input
+              id="passwordCheck"
+              placeholder="비밀번호와 일치하는 값을 입력해주세요"
+              type={passCheckShow ? "text" : "password"}
+              className={errors.passwordCheck ? "err" : ""}
+              aria-invalid={errors.passwordCheck ? "true" : "false"}
+              {...(register("passwordCheck"), passCheckRegister)}
```

Comment:
passCheckRegister 가 어떤 건지 예상이 안됩니다.
선언부 확인해보니 객체를 통으로 묶지 않고 아래처럼 수정하면 좋을 것 같아요.


```suggestion
              {...register("passwordCheck")}
              onBlur={(e) => {
                validatePasswordCheck(e);
                register.onBlur(e);
              }}
```